generator/statistics.cpp

```diff
@@ -55,20 +60,50 @@ namespace stats
     {
       f.ParseBeforeStatistic();
 
-      FeatureType::InnerGeomStat const innerStats = f.GetInnerStatistic();
+      FeatureType::InnerGeomStat const innerStats = f.GetInnerStats();
 
       m_info.m_inner[0].Add(innerStats.m_points);
       m_info.m_inner[1].Add(innerStats.m_strips);
       m_info.m_inner[2].Add(innerStats.m_size);
 
-      // get geometry size for the best geometry
-      FeatureType::GeomStat const geom = f.GetGeometrySize(FeatureType::BEST_GEOMETRY);
-      FeatureType::GeomStat const trg = f.GetTrianglesSize(FeatureType::BEST_GEOMETRY);
+      // Get size stats and load the best geometry.
+      FeatureType::GeomStat const geom = f.GetOuterGeometryStats();
+      FeatureType::GeomStat const trg = f.GetOuterTrianglesStats();
+
+      uint32_t geomSize = 0, trgSize = 0;
+      int const n = feature::DataHeader::kMaxScalesCount;
+      for (int ind = 0; ind < n; ++ind)
+      {
+        m_info.m_byLineGeom[ind].Add(geom.m_sizes[ind], geom.m_elements[ind]);
```

Comment:
Please make reusable constants for each call to geom.m_sizes[i] and alike, it will make the code more readable.